codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/deprecatedrenamerequest.java

```diff
@@ -0,0 +1,264 @@
+        return (obj, val) -> s.accept((Builder) obj, val);
+    }
+
+    public interface Builder extends JsonProtocolTestsRequest.Builder, SdkPojo, CopyableBuilder<Builder, DeprecatedRenameRequest> {
+        /**
+         * Sets the value of the NewNameNoDeprecation property for this object.
+         *
+         * @param newNameNoDeprecation
+         *        The new value for the NewNameNoDeprecation property for this object.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder newNameNoDeprecation(String newNameNoDeprecation);
+
+        /**
+         * Sets the value of the NewName property for this object.
+         *
+         * @param newName
+         *        The new value for the NewName property for this object.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        Builder newName(String newName);
+
+        /**
+         * Sets the value of the NewName property for this object.
+         *
+         * @param newName
+         *        The new value for the NewName property for this object.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         * @deprecated Use {@link #newName(String)}
+         */
+        @Deprecated
+        Builder originalNameDeprecated(String newName);
+
+        @Override
+        Builder overrideConfiguration(AwsRequestOverrideConfiguration overrideConfiguration);
+
+        @Override
+        Builder overrideConfiguration(Consumer<AwsRequestOverrideConfiguration.Builder> builderConsumer);
+    }
+
+    static final class BuilderImpl extends JsonProtocolTestsRequest.BuilderImpl implements Builder {
+        private String newNameNoDeprecation;
+
+        private String newName;
+
+        private BuilderImpl() {
+        }
+
+        private BuilderImpl(DeprecatedRenameRequest model) {
+            super(model);
+            newNameNoDeprecation(model.newNameNoDeprecation);
+            newName(model.newName);
+        }
+
+        public final String getNewNameNoDeprecation() {
+            return newNameNoDeprecation;
+        }
+
+        @Override
+        public final Builder newNameNoDeprecation(String newNameNoDeprecation) {
+            this.newNameNoDeprecation = newNameNoDeprecation;
+            return this;
+        }
+
+        public final void setNewNameNoDeprecation(String newNameNoDeprecation) {
+            this.newNameNoDeprecation = newNameNoDeprecation;
+        }
+
+        public final String getNewName() {
+            return newName;
+        }
+
+        @Override
+        public final Builder newName(String newName) {
+            this.newName = newName;
+            return this;
+        }
+
+        @Override
+        public final Builder originalNameDeprecated(String newName) {
+            this.newName = newName;
+            return this;
+        }
```

Comment:
Minor Nit: Should probably have bean-based methods for the deprecated field. These are around for people using things like spring-xml, so it's likely not going to break anyone to exclude them, but it'd be nice to have for completeness.

Reply:
I'll try to have this for you by Monday.

Reply:
I think this is good now.